include/deal.II/distributed/tria_base.h

```diff
@@ -398,6 +398,455 @@ namespace parallel
+     * object that now is the time when they are ready by calling the
+     * current function.
+     *
+     * The supplied callback function is then called for each newly locally
+     * owned cell. The first argument to the callback is an iterator that
+     * designates the cell; the second argument indicates the status of the
+     * cell in question; and the third argument localizes a memory area by
+     * two iterators that contains the data that was previously saved from
+     * the callback provided to register_data_attach().
+     *
+     * The CellStatus will indicate if the cell was refined, coarsened, or
+     * persisted unchanged. The @p cell_iterator argument to the callback
+     * will then either be an active,
+     * locally owned cell (if the cell was not refined), or the immediate
+     * parent if it was refined during execute_coarsening_and_refinement().
+     * Therefore, contrary to during register_data_attach(), you can now
+     * access the children if the status is `CELL_REFINE` but no longer for
+     * callbacks with status `CELL_COARSEN`.
+     *
+     * The first argument to this function, `handle`, corresponds to
+     * the return value of register_data_attach(). (The precise
+     * meaning of what the numeric value of this handle is supposed
+     * to represent is neither important, nor should you try to use
+     * it for anything other than transmit information between a
+     * call to register_data_attach() to the corresponding call to
+     * notify_ready_to_unpack().)
+     */
+    void
+    notify_ready_to_unpack(
+      const unsigned int handle,
+      const std::function<
+        void(const cell_iterator &,
+             const CellStatus,
+             const boost::iterator_range<std::vector<char>::const_iterator> &)>
+        &unpack_callback);
+
+    /**
+     * Go through all cells and store the relations between locally
+     * owned quadrants and cells in the private member
+     * local_cell_relations.
+     *
+     * The stored vector will be ordered by the occurrence of quadrants.
+     */
+    virtual void
+    update_cell_relations() = 0;
+
+    /**
+     * This auxiliary data structure stores the relation between
+     * a deal.II cell and its current CellStatus. For an extensive
+     * description of the latter, see the documentation for the member
+     * function register_data_attach().
+     */
+    using cell_relation_t = typename std::tuple<CellStatus, cell_iterator>;
+
+    /**
+     * Vector of tuples, which each contain a deal.II cell
+     * and their relation after refinement. To update its contents, use the
+     * compute_cell_relations member function.
+     *
+     * The size of this vector is assumed to be equal to the number of locally
+     * owned quadrants in the parallel_forest object.
+     */
+    // TODO: convert local_cell_relations to a more efficient container (maybe
+    // std::map<cell_iterator, CellStatus>?).
```

Comment:
Don't leave `TODO`s in the code. They will be easily overlooked in the future.

Rather open a new issue that will suggest to use a more efficient container here, and link it to this PR.

Reply:
Done! (#11589)